contracts/database/orm/orm.go

```diff
@@ -141,6 +141,8 @@ type Query interface {
 	Where(query any, args ...any) Query
 	// WhereIn adds a "where column in" clause to the query.
 	WhereIn(column string, values []any) Query
+	// WhereHas applies a subquery condition to filter results based on a related table.
+	WhereHas(parentModel any, childModel any,conditions func(Query) (Query,error) ) (Query,error)
```

Comment:
I suggest that the parameters are better be passed like Laravel: 

```
$posts = Post::whereHas('comments', function (Builder $query) {
    $query->where('content', 'like', 'code%');
})->get();
```

Therefore, the `WhereHas` can be:

```
facades.Orm().Query().WhereHas("Comments", func(query ormcontract.Query) (ormcontract.Query, error) {
     modifiedQuery := query.Where("Name = ?", "Book A")
     return modifiedQuery, nil
}).Get(&users)
```